ion/textreader.go

```diff
@@ -425,6 +459,31 @@ func (t *textReader) onSymbol(val string, tok token, ws bool) error {
 	return nil
 }
 
+// readSymbol reads a text and returns a symbol token.
```

Comment:
Could you flesh out this comment a bit?

Reply:
From the code this looks like a function that can take a SID as a Go `string` and use it as a SID, i.e., `readSymbol("$42")`. If that is indeed the intent, maybe call this `readAsSID`?

Reply:
> From the code this looks like a function that can take a SID as a Go string and use it as a SID, i.e., readSymbol("$42"). If that is indeed the intent, maybe call this readAsSID?

The function reads SID text as well as symbol text e.g., `foo` and return a symboltoken.

Updated comment to:
```
// readSymbol reads a text as either a symbol identifier e.g., "$42" or a symbol text e.g., "foo" and resolves to a
// symbol token.
```